picard/cluster.py

```diff
@@ -115,18 +115,22 @@ def _update_related_album(self, added_files=None, removed_files=None):
             self.related_album.update()
 
     def add_files(self, files):
+        added_files = []
```

Comment:
It would be better to use a set for this, it is significantly faster for doing `file in self.files`. Does order matter?

Reply:
I assumed order would matter for a cluster, but maybe it doesn't. I need to check

Reply:
Cluster files are sorted in https://github.com/metabrainz/picard/blob/master/picard/tagger.py#L823

So even without column sorting they show up in order. I think generally changing `Cluster.files` to a set could also have some side effects or break someone's plugin. Otherwise it would work in general, I tested it.

Making `added_files` a set would not make much difference I think, we don't search in it. It's just a list that gets appended to.

We could in this method do a `existing_files= set(self.files)` call, and then use existing_files in `if file in existing_files`:

```
def add_files(self, files):
    added_files = []
    existing_files= set(self.files)
    for file in files:
        if file in existing_files:
            continue
        added_files.append(file)
        ...
```

But I doubt there is much difference.

Reply:
Profile it ;) So we can decide.